apps/api/src/tests/services/application.ts

```diff
@@ -0,0 +1,197 @@
+ * OF MERCHANTABILITY AND FITNESS FOR A PARTICULAR PURPOSE ARE DISCLAIMED. IN NO EVENT
+ * SHALL THE COPYRIGHT HOLDER OR CONTRIBUTORS BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT NOT LIMITED
+ * TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE, DATA, OR PROFITS;
+ * OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY THEORY OF LIABILITY, WHETHER
+ * IN CONTRACT, STRICT LIABILITY, OR TORT (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN
+ * ANY WAY OUT OF THE USE OF THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+import assert from 'node:assert';
+import { after, before, describe, it } from 'node:test';
+
+import { PostgreSqlContainer, StartedPostgreSqlContainer } from '@testcontainers/postgresql';
+
+import { ApplicationStates } from 'pcgl-daco/packages/data-model/src/types.ts';
+import { initMigration, startDb, type PostgresDb } from '../../db/index.ts';
+import { applications } from '../../db/schemas/applications.ts';
+import service from '../../service/application-service.ts';
+
+const PG_DATABASE = process.env.PG_DATABASE || 'testUser';
+const PG_USER = process.env.PG_USER || 'testPassword';
+const PG_PASSWORD = process.env.PG_PASSWORD || 'postgres';
+
+describe('Application Service', () => {
+	let db: PostgresDb;
+	let applicationService: ReturnType<typeof service>;
+	let container: StartedPostgreSqlContainer;
+
+	const user_id = 'testUser@oicr.on.ca';
+
+	before(async () => {
+		container = await new PostgreSqlContainer()
+			.withUsername(PG_USER)
+			.withPassword(PG_PASSWORD)
+			.withDatabase(PG_DATABASE)
+			.start();
+
+		const connectionString = container.getConnectionUri();
+		db = startDb(connectionString);
+
+		await initMigration(db);
+
+		applicationService = service(db);
+		// TODO: create file with seed data for postgres to test read actions
+		// Not yet supported in Drizzle Kit:
+		// https://orm.drizzle.team/docs/kit-seed-data
+	});
+
+	it('should create applications with status DRAFT and submitted user_id', async () => {
+		const application = await applicationService.createApplication({ user_id });
+
+		assert.notEqual(application, null);
+		assert.strictEqual(application?.user_id, user_id);
+		assert.strictEqual(application?.state, ApplicationStates.DRAFT);
+	});
+
+	it('should get applications requested by id, with application_contents', async () => {
+		const applicationRecords = await applicationService.listApplications({ user_id });
+
+		assert.ok(Array.isArray(applicationRecords));
+
+		const { id } = applicationRecords[0];
+
+		const requestedApplication = await applicationService.getApplicationById({ id });
+
+		assert.notEqual(requestedApplication, null);
+		assert.strictEqual(requestedApplication?.id, id);
+		assert.strictEqual(requestedApplication?.id, requestedApplication?.contents?.application_id);
+	});
+
+	it('should list applications filtered by user_id', async () => {
```

Comment:
The next set of tests are for the listApplications, can we group them into a `suite('listApplications', async ()=>{...})`?

Reply:
Yes, added a `describe()` which is just alias for suite, later we can add same for create/get etc.